### PR TITLE
fix(schematics): make override schematic robust to path seperators

### PR DESCRIPTION
**Pull request by MaxKless**
Currently, the override schematic only accepts paths with forward slashes.

Now, it also accepts paths with backslashes.

[AB#76433](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/76433)

### CODE REVIEW

File: schematics/src/helpers/override/factory.ts [hunk 1]
```diff
@@ -19,9 +20,15 @@ export function override(options: Options): Rule {
     const workspace = await getWorkspace(host);
     const project = workspace.projects.get(options.project);
     const sourceRoot = project.sourceRoot;
-    const from = `${
-      options.path ? `${options.path}/` : !options.from?.startsWith(`${sourceRoot}/app/`) ? `${sourceRoot}/app/` : ''
-    }${options.from.replace(/\/$/, '')}`;
+    const from = normalize(
+      `${
+        options.path
```
Comment (dhhyi) [inline schematics/src/helpers/override/factory.ts:25]: I would not use path sep and stick to slashes only.
just do `normalize(options.path).replace(/\\/, '/')` and keep the remaining.

File: schematics/src/helpers/override/factory_spec.ts [hunk 2]
```diff
@@ -282,4 +282,19 @@ describe('override Schematic', () => {
       ]
     `);
   });
+
+  it('should override file if path contains backslashes', async () => {
+    const tree = await runOverride({
+      from: 'core\\routing\\product.route.ts',
```
Comment (dhhyi) [inline schematics/src/helpers/override/factory_spec.ts:288]: On win i always have paths also starting with `.\`.

**Review by dhhyi — CHANGES_REQUESTED** [review]

File: schematics/src/helpers/override/factory.ts [hunk 3]
```diff
@@ -20,15 +20,14 @@ export function override(options: Options): Rule {
...
-        options.path
-          ? `${options.path}${sep}`
-          : !options.from?.startsWith(`${sourceRoot}${sep}app${sep}`)
-          ? `${sourceRoot}${sep}app${sep}`
-          : ''
-      }${options.from.replace(/\/$/, '')}`
-    );
+    const path = normalize(options.path ?? '')
+      .replace(/\\/g, '/')
+      .replace(/^\./, '');
```
Comment (dhhyi) [inline schematics/src/helpers/override/factory.ts:25]: `normalize` _should_ already take care of that.

**Review by dhhyi — CHANGES_REQUESTED** [review]

File: schematics/src/helpers/override/factory_spec.ts (hunk 2 above)
Comment (MaxKless) [reply schematics/src/helpers/override/factory_spec.ts:288]: also works with those paths now.

File: schematics/src/helpers/override/factory.ts (hunk 3 above)
Comment (dhhyi) [reply schematics/src/helpers/override/factory.ts:25]: @MaxKless Am I wrong here?

**Pull request merged**